File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLUserProfilePhotoEmpty.java
package com.github.badoualy.telegram.tl.api;

import static com.github.badoualy.telegram.tl.StreamUtils.*;

import java.lang.Override;
import java.lang.String;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLUserProfilePhotoEmpty extends TLAbsUserProfilePhoto {
    public static final int CLASS_ID = 0x4f11bae1;

    public TLUserProfilePhotoEmpty() {
    }

    @Override
    public String toString() {
        return "userProfilePhotoEmpty#4f11bae1";
    }

    @Override
    public int getClassId() {
        return CLASS_ID;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLUserProfilePhotoEmpty.kt
package com.github.badoualy.telegram.tl.api

/**
 * userProfilePhotoEmpty#4f11bae1
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLUserProfilePhotoEmpty : TLAbsUserProfilePhoto() {
    private val _constructor: String = "userProfilePhotoEmpty#4f11bae1"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLUserProfilePhotoEmpty) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x4f11bae1.toInt()
    }
}
